### PR TITLE
Bitcoin SPV relayer: Relays blocks from Bitcoin to Native

### PR DESCRIPTION
**Pull request by udit-gulati**
## Description

Bitcoin -> Native blocks relayer

closes: #XXXX

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary

### CODE REVIEW

File: reporter/block_handler.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
+package reporter
+
+import (
+	"fmt"
+
+	"github.com/babylonchain/vigilante/types"
```
Comment (robert-zaremba) [inline reporter/block_handler.go:6]: why do we import babylon here?

File: reporter/expected_babylon_client.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	"context"
+
+	"github.com/babylonchain/babylon/client/config"
+	btcctypes "github.com/babylonchain/babylon/x/btccheckpoint/types"
+	btclctypes "github.com/babylonchain/babylon/x/btclightclient/types"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	pv "github.com/cosmos/relayer/v2/relayer/provider"
+)
+
+type BabylonClient interface {
```
Comment (robert-zaremba) [inline reporter/expected_babylon_client.go:13]: why we need a babylon client?

**Review by robert-zaremba — COMMENTED** [review]
we need to change the structure of this code
- we should not have generic `types` directory
- main package is missing - what's the executable?
- we should rename reporter to `bitcoin`

File: reporter/block_handler.go (hunk 1 above)
Comment (udit-gulati) [reply reporter/block_handler.go:6]: I've left a few types imported from babylon cause they are part of checkpointing logic which is to be removed, so can test it for now without too much change to the forked code and remove it at the end of testing.

**Comment by robert-zaremba** [comment]
Planning; discussing deadline to finish this task (including reviews) by Dec 13

File: btcclient/query.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+		c.logger.Debug(
+			"failed to query the block verbose", zap.String("hash", hash.String()), zap.Error(err))
+	}
+
+	return blockVerbose, nil
+}
+
+// getChainBlocks returns a chain of indexed blocks from the block at baseHeight to the tipBlock
+// note: the caller needs to ensure that tipBlock is on the blockchain
+func (c *Client) getChainBlocks(baseHeight uint64, tipBlock *types.IndexedBlock) ([]*types.IndexedBlock, error) {
```
Comment (vuvoth) [inline btcclient/query.go:145]: I just want to confirm this is a wrapper of bitcoin RPC.

File: btcclient/testutils.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+import (
+	"time"
+
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/gonative-cc/relayer/reporter/config"
+	"github.com/gonative-cc/relayer/reporter/types"
+)
+
+func NewTestClientWithWsSubscriber(rpcClient *rpcclient.Client, cfg *config.BTCConfig, retrySleepTime time.Duration, maxRetrySleepTime time.Duration, blockEventChan chan *types.BlockEvent) (*Client, error) {
```
Comment (vuvoth) [inline btcclient/testutils.go:11]: If this is for testing only, why do we need to export this?

**Review by vuvoth — COMMENTED** [review]
Quick review!

File: btcclient/query.go (hunk 3 above)
Comment (udit-gulati) [reply btcclient/query.go:145]: Yes, this is a wrapper

**Pull request closed without merging**